### ISSUE TITLE
Error: Status code 404 upon checking for videos

### ISSUE THREAD

**Issue by NightMean** [MarcelRobitaille/bbyen#22]
Ho ho ho, I guess I made it to Santa’s naughty list this year. While everyone else is unwrapping shiny new gifts, I’m here unwrapping this issue :)

I have added couple of new channels in the past days as I had more time during the christmas time but it seems like bbyen didn't like that. It seems to me that something (a channel?) got deleted before it could be checked? I'm not sure. 

Here is my latest config with the credentials hidden: 
[config.json](https://github.com/user-attachments/files/18256027/config.json)

Here is the sqlite database in case you need it for troubleshooting. 
[database.zip](https://github.com/user-attachments/files/18256041/database.zip)

Running latest version up to 22f190cddbfdf2b506c2454173ca20a859b021b4 commit

```
npm[3104578]: 2024-12-26T22:32:19.826Z [verbose] [videos]: 268/408 Checking channel Make It Work (UCGoreZKPBtCXCf54F3DF4ug)
npm[3104578]: 2024-12-26T22:32:19.866Z [error] [videos]: Status code 404
npm[3104578]: Error: Status code 404
npm[3104578]:     at ClientRequest.<anonymous> (/opt/youtubenotify-bbyen/node_modules/rss-parser/lib/parser.js:88:25)
npm[3104578]:     at Object.onceWrapper (node:events:476:26)
npm[3104578]:     at ClientRequest.emit (node:events:369:20)
npm[3104578]:     at ClientRequest.emit (node:domain:470:12)
npm[3104578]:     at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:646:27)
npm[3104578]:     at HTTPParser.parserOnHeadersComplete (node:_http_common:129:17)
npm[3104578]:     at HTTPParser.execute (<anonymous>)
npm[3104578]:     at TLSSocket.socketOnData (node:_http_client:512:22)
npm[3104578]:     at TLSSocket.emit (node:events:369:20)
npm[3104578]:     at TLSSocket.emit (node:domain:470:12)
```

**Comment by MarcelRobitaille** [comment]
Hi @NightMean. Always good to hear from you

> Ho ho ho, I guess I made it to Santa’s naughty list this year. While everyone else is unwrapping shiny new gifts, I’m here unwrapping this issue :)

:rofl: :rofl: :rofl: 

Does the problem happen repeatably? I was not able to reproduce it with your config. The stack trace you gave me only shows library code. I don't know where that is called from my code.

**Comment by NightMean** [comment]
> Does the problem happen repeatably? I was not able to reproduce it with your config. The stack trace you gave me only shows library code. I don't know where that is called from my code.

Yes. It happens everytime bbyen checks for new videos. It can check half of the channels but it stops everytime it gets to the particular channel which errors out and stops going further. It doesn't crash though. 
I have not tried re-authing but I don't expect that will solve it.

**Comment by NightMean** [comment]
As expected, even after doing a reauth I got the same error.

**Comment by NightMean** [comment]
I have dug deeper into it and found the culprit. It's not the next channel that can't be checked but the Make It Work as shown above. 
The issue is that the channel was recently hacked, all of the videos have been deleted and name was changed. I assume the creator managed to get in touch with Youtube support to lock the channel for now or something like that. 

Currently, bbyen throws 404 error on it's channel which you should also see here: https://www.youtube.com/feeds/videos.xml?channel_id=UCGoreZKPBtCXCf54F3DF4ug

So, the question is how bbyen should handle this type of issue. Perhaps it could be also treated as a deleted channel in the database and upon channel checks in the future it could check if it was not restored. 

Although I'm wondering why did it work for you with my config.

**Comment by MarcelRobitaille** [comment]
I'm sorry. I made a mistake reproducing it. I'm now able to.

Do you have a link to the new channel? If the channel was deleted and recreated, I don't know what BBYEN can do.

I changed the code a bit so it prints a warning and continues:
```
2024-12-31T13:20:28.868Z [warn] [videos]: Error parsing videos from 'https://www.youtube.com/feeds/videos.xml?channel_id=UCGoreZKPBtCXCf54F3DF4ug': Status code 404
2024-12-31T13:20:28.868Z [info] [videos]: Finished checking for new video
```

**Comment by NightMean** [comment]
> Do you have a link to the new channel? If the channel was deleted and recreated, I don't know what BBYEN can do.

I don't as I expect the channel will be recovered sometime during next year. Of course I don't want bbyen to search for the new channel with the same name 😄 

> I changed the code a bit so it prints a warning and continues: 

Does this mean that in case the channel would be recovered in future, bbyen would catch this and check the channel again? If so, it's basically perfect.

**Comment by MarcelRobitaille** [comment]
> Does this mean that in case the channel would be recovered in future, bbyen would catch this and check the channel again? If so, it's basically perfect.

Yes, that's correct. If you're happy with that, I'll merge it. I also thought about some kind of error reporting by email, but for now I think this solution is fine.

**Comment by NightMean** [comment]
Sounds fine for me :) I don't think the email reporting is really necessary. It can be seen in the logs so that's good enough for me. 
Thanks!

**Comment by MarcelRobitaille** [comment]
Implemented in 3.0.2
